examples/Controllers/Latching/optimalTimeCalc.m

```diff
@@ -0,0 +1,74 @@
+% This script identifies the dynamics of the float in the respective wave 
+% conditions and determines the optimal proportional gain value for a 
+% passive controller (for regular waves)
+
+close all; clear all; clc;
+
+% Inputs (from wecSimInputFile)
+simu = simulationClass();
+body(1) = bodyClass('../hydroData/rm3.h5');
+waves.height = 2.5;
+waves.period = 9.52;
+
+% Load hydrodynamic data for float from BEM
+floatHydro = readBEMIOH5(body.h5File, 1, body.meanDrift);
+
+% Define wave conditions
+H = waves.height;
+A = H/2;
+T = waves.period;
+omega = (1/T)*(2*pi);
+
+% Define excitation force based on wave conditions
+ampSpect = zeros(length(floatHydro.simulation_parameters.w),1);
+[~,closestIndOmega] = min(abs(omega-floatHydro.simulation_parameters.w));
+ampSpect(closestIndOmega) = A;
+FeRao = squeeze(floatHydro.hydro_coeffs.excitation.re(3,:))'*simu.rho*simu.gravity + squeeze(floatHydro.hydro_coeffs.excitation.im(3,:))'*simu.rho*simu.gravity*1j;
+Fexc = ampSpect.*FeRao;
+
+% Define the intrinsic mechanical impedance for the device
+mass = simu.rho*floatHydro.properties.volume;
+addedMass = squeeze(floatHydro.hydro_coeffs.added_mass.all(3,3,:))*simu.rho;
```

Comment:
Similar to my above comment, but if we are hard coding these then I think it needs to be an applications case.

Reply:
Good point, I can work on switching the float out with IEA OES Task 10 sphere to be more clear.